.github/workflows/go-check.yaml

```diff
@@ -0,0 +1,47 @@
+name: Go related checks
```

Comment:
Nit: This should be `Go-related checks`. No need to send an update just for that though :-)